src/commands/logs.rs

```diff
@@ -0,0 +1,168 @@
+use std::ops::Sub;
+use std::time::Duration;
+
+use anyhow::{anyhow, Result};
+use chrono::Utc;
+use cloud_openapi::models::Entry;
+
+use cloud::{client::Client as CloudClient, CloudClientExt, CloudClientInterface};
+
+use crate::commands::create_cloud_client;
+use crate::commands::deploy::SPIN_DEPLOY_CHANNEL_NAME;
+use crate::opts::*;
+use clap::Parser;
+use uuid::Uuid;
+
+/// Fetch and tail logs of an application deployed to the Fermyon Cloud.
+#[derive(Parser, Debug)]
+#[clap(about = "fetch logs for an app from Fermyon Cloud")]
+pub struct LogsCommand {
+    /// Find app to fetch logs for in the Fermyon instance saved under the specified name.
+    /// If omitted, Spin looks for app in default unnamed instance.
+    #[clap(
+        name = "environment-name",
+        long = "environment-name",
+        env = DEPLOYMENT_ENV_NAME_ENV
+    )]
+    pub deployment_env_id: Option<String>,
+
+    /// App name
+    #[clap(name = "app-name", long = "app-name")]
+    pub app_name: String,
+
+    /// Follow logs output
+    #[clap(name = "follow", long = "follow")]
+    pub follow: bool,
+
+    /// Number of lines to show from the end of the logs
+    #[clap(name = "tail", long = "tail", default_value = "10")]
+    pub max_lines: i32,
+
+    /// interval in secs to refresh logs from cloud
+    #[clap(parse(try_from_str = parse_interval), name="interval", long="interval", default_value = "2")]
+    pub interval_secs: u64,
+
+    /// fetch logs since
```

Comment:
If I read this as a doc string I would think I should enter a date or time.  "Fetch logs since 2:30."  Provide guidance e.g.
```suggestion
    /// How long ago to fetch logs from, expressed as a duration. The duration format
    /// is a number and a unit, where the unit is 's' for seconds, 'm' for minutes, 'h' for hours
    /// or 'd' for days (e.g. "30m" for 30 minutes ago).  The default it 7 days.
```

(The start of that suggestion feels a bit too wordy and indirect, but hopefully gives an idea of the kind of thing.)

Reply:
```
Only return logs newer than a relative duration.
```